### PR TITLE
Add exploratory figures for bail.csv

### PR DESCRIPTION
**Pull request by trangdata**

### CODE REVIEW

File: analyses/team3/explore-bails.Rmd [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+```{r message=FALSE, warning=FALSE}
+library(tidyverse)
+library(skimr)
+library(lubridate)
+library(ggradar)
+
+colorBlindGrey6 <- c("#0072B2", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#CC79A7")
+
+theme_set(theme_minimal())
+theme_update(panel.grid.minor = element_blank())
```
Comment (rjake) [inline analyses/team3/explore-bails.Rmd:32]: where has this been all my life 😮

File: analyses/team3/explore-bails.Rmd [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+# end_period_2 <- larry_date + period(2, 'years')
+# 
+# bail_larry <- bail_short %>%
+#   mutate(period = case_when(
+#     action_date < larry_date & action_date >= start_period_1 ~ 'before',
+#     action_date >= larry_date & action_date < end_period_2 ~ 'after',
+#     TRUE ~ NA_character_)) %>%
+#   filter(!is.na(period))
+# 
+# save(bail_larry, file = 'processed-data/bail-larry.rds')
```
Comment (rjake) [inline analyses/team3/explore-bails.Rmd:53]: I like to put this at the end of the code after a `knit_exit()` chunk. The `eval=FALSE` isn't required but I like to keep it there so others know the code doesn't run
```r
    Your actual code...

    ```{r}
    knit_exit()
    ```

    Nothing after this point runs when knit

    # Data Prep
    ```{r, eval=FALSE}
    this code doesn't run
    ```


```

File: analyses/team3/explore-bails.Rmd [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+```{r fig.width=10}
+bail_judges <- bail_larry %>%
+  mutate(judge_name = case_when(
+    is.na(participant_name__first_name) ~ NA_character_,
+    TRUE ~  paste(participant_name__first_name,
+                            participant_name__last_name)))
+
+judges <- bail_judges %>%
+  count(period, docket_id, judge_name) %>% 
+  filter(!is.na(judge_name)) %>%
```
Comment (rjake) [inline analyses/team3/explore-bails.Rmd]: you can also use this
```suggestion
  drop_na(judge_name) %>% 
```

File: analyses/team3/explore-bails.Rmd [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  ungroup() %>% 
+  group_by(period, judge_name) %>% 
+  summarise(total_adj_n = sum(weighted_n)) 
+
+top_judges <- judges %>% 
+  top_n(10, total_adj_n) %>% 
+  pull(judge_name)
+
+judges %>% 
+  filter(judge_name %in% top_judges) %>%
```
Comment (rjake) [inline analyses/team3/explore-bails.Rmd]: Can you do this in one go?

File: analyses/team3/explore-bails.Rmd [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+              values_from = total_adj_n, 
+              values_fill = 0) %>% 
+  ggradar(values.radar = c(NA, '2,500', '5,000'),
+        group.colours = colorBlindGrey6,
+        grid.mid = 2500,
+        grid.max = 5000,
+        base.size = 1,
+        legend.position = "top") + 
+  labs(title = 'Top judges involvement in bails') +
+  guides(color = guide_legend(reverse = TRUE))
```
Comment (rjake) [inline analyses/team3/explore-bails.Rmd:94]: Nice, didn't know about this one!

File: analyses/team3/explore-bails.Rmd [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+- Number of defendants who fail to pay low monetary bail
+- Trends in case dismissals, probation, parole, etc.
+
+```{r message=FALSE, warning=FALSE}
+library(tidyverse)
+library(skimr)
+library(lubridate)
+library(ggradar)
+
+colorBlindGrey6 <- c("#0072B2", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#CC79A7")
```
Comment (rjake) [inline analyses/team3/explore-bails.Rmd]: 👀 Good looking out! Why "grey" in the name? Also, could this be 🐍 over 🐫 case?

File: analyses/team3/explore-bails.Rmd [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+
+Changes in judges involvement:
+
+```{r fig.width=10}
+bail_judges <- bail_larry %>%
+  mutate(judge_name = case_when(
+    is.na(participant_name__first_name) ~ NA_character_,
+    TRUE ~  paste(participant_name__first_name,
+                            participant_name__last_name)))
```
Comment (rjake) [inline analyses/team3/explore-bails.Rmd:68]: Not sure if this helps at all but I just learned about `na.omit()` if you don't want the NA to come in as a string with `paste()`:
```r
paste(
  na.omit(participant_name__first_name),
  na.omit( participant_name__last_name)
)

**Review by rjake — APPROVED** [review]
Great job 🏆 I had a few comments to consider but overall looks good 😃

File: analyses/team3/explore-bails.Rmd (hunk 3 above)
Comment (trangdata) [reply analyses/team3/explore-bails.Rmd]: Ahhh I knew about this. `drop_na()` scares me somehow but I should start using it more. Adding suggestion to batch!

File: analyses/team3/explore-bails.Rmd (hunk 7 above)
Comment (trangdata) [reply analyses/team3/explore-bails.Rmd:68]: Hmm thanks I'll investigate this.

File: analyses/team3/explore-bails.Rmd (hunk 4 above)
Comment (trangdata) [reply analyses/team3/explore-bails.Rmd]: Yes. To be fixed.

Comment (trangdata) [reply analyses/team3/explore-bails.Rmd]: Discussed via Slack.

File: analyses/team3/explore-bails.Rmd (hunk 6 above)
Comment (trangdata) [reply analyses/team3/explore-bails.Rmd]: ah yes I copy-pasted this from somewhere. Will fix.

File: analyses/team3/explore-bails.Rmd (hunk 2 above)
Comment (trangdata) [reply analyses/team3/explore-bails.Rmd:53]: Addressed in https://github.com/rjake/2021_datathon/pull/4/commits/2b4ff41dc6ebb9327bf24a6a120191d9672bb0bc.

File: analyses/team3/explore-bails.Rmd (hunk 6 above)
Comment (trangdata) [reply analyses/team3/explore-bails.Rmd]: Addressed in https://github.com/rjake/2021_datathon/pull/4/commits/2b4ff41dc6ebb9327bf24a6a120191d9672bb0bc

File: analyses/team3/explore-bails.Rmd (hunk 4 above)
Comment (trangdata) [reply analyses/team3/explore-bails.Rmd]: Addressed in https://github.com/rjake/2021_datathon/pull/4/commits/975c320b948eaa6ac37a0486372fdd3d2c016531.

File: analyses/team3/explore-bails.Rmd (hunk 7 above)
Comment (trangdata) [reply analyses/team3/explore-bails.Rmd:68]: So I looked into this a bit, and I think na.omit() would be quite dangerous inside mutate(). It omits all NAs for each column, then combines and loops over the shorter column after removing NAs.

``` r
> bigrams_filtered %>% 
+   slice(1:10) %>% 
+   mutate(new_bigram = 
+            paste(na.omit(word1), word2))
# A tibble: 10 x 4
   book                word1   word2  new_bigram    
   <fct>               <chr>   <chr>  <chr>         
 1 Sense & Sensibility NA      NA     jane NA       
 2 Sense & Sensibility jane    austen chapter austen
 3 Sense & Sensibility NA      NA     jane NA       
 4 Sense & Sensibility NA      NA     chapter NA    
 5 Sense & Sensibility NA      NA     jane NA       
 6 Sense & Sensibility NA      NA     chapter NA    
 7 Sense & Sensibility NA      NA     jane NA       
 8 Sense & Sensibility NA      NA     chapter NA    
 9 Sense & Sensibility chapter 1      jane 1        
10 Sense & Sensibility NA      NA     chapter NA    
```

Comment (rjake) [reply analyses/team3/explore-bails.Rmd:68]: whoa, what?!?!? I did not know 😲

**Pull request merged**